File: app/src/real/java/com/vsa/filmoteca/data/repository/MoviesDataRepository.java
package com.vsa.filmoteca.data.repository;

import com.vsa.filmoteca.data.domain.Movie;
import com.vsa.filmoteca.data.domain.mapper.DetailHtmlParser;
import com.vsa.filmoteca.data.domain.mapper.MovieHtmlMapper;
import com.vsa.filmoteca.data.repository.ws.CacheRequestInterceptor;
import com.vsa.filmoteca.data.repository.ws.WSClient;

import java.util.List;

import rx.Observable;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

/**
 * Created by albertovecinasanchez on 7/12/15.
 */
public class MoviesDataRepository {

    private static MoviesDataRepository sRepository;

    public Observable<List<Movie>> moviesList() {
        return WSClient.getClient(CacheRequestInterceptor.CachePolicy.PRIORITY_NETWORK).moviesListHtml().map(MovieHtmlMapper::transformMovie)
                .subscribeOn(Schedulers.newThread())
                .observeOn(AndroidSchedulers.mainThread());
    }

    public Observable<String> movieDetail(String url) {
        return WSClient.getClient(CacheRequestInterceptor.CachePolicy.PRIORITY_NETWORK).movieDetail(url).map(html -> {
            try {
                return DetailHtmlParser.parse(html);
            } catch (Exception e) {
                e.printStackTrace();
                return "";
            }
        }).subscribeOn(Schedulers.newThread())
                .observeOn(AndroidSchedulers.mainThread());
    }

}

File: app/src/real/java/com/vsa/filmoteca/data/repository/MoviesDataRepository.kt
package com.vsa.filmoteca.data.repository

import com.vsa.filmoteca.data.domain.Movie
import com.vsa.filmoteca.data.domain.mapper.DetailHtmlParser
import com.vsa.filmoteca.data.domain.mapper.MovieHtmlMapper
import com.vsa.filmoteca.data.repository.ws.FilmotecaInterface
import rx.Observable
import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers
import javax.inject.Inject

/**
 * Created by albertovecinasanchez on 7/12/15.
 */
class MoviesDataRepository @Inject constructor(private val filmotecaInterface: FilmotecaInterface) {

    fun moviesList(): Observable<List<Movie>> {
        return filmotecaInterface.moviesListHtml().map { MovieHtmlMapper.transformMovie(it) }
                .subscribeOn(Schedulers.newThread())
                .observeOn(AndroidSchedulers.mainThread())
    }

    fun movieDetail(url: String): Observable<String> {
        return filmotecaInterface.movieDetail(url).map<String> { html ->
            try {
                DetailHtmlParser.parse(html)
            } catch (e: Exception) {
                e.printStackTrace()
                ""
            }
        }.subscribeOn(Schedulers.newThread())
                .observeOn(AndroidSchedulers.mainThread())
    }

}
